crates/vizia_baseview/src/application.rs

```diff
@@ -207,26 +208,17 @@ impl ApplicationRunner {
             self.current_user_scale_factor = cx.user_scale_factor();
 
             // The user scale factor is not part of the HiDPI scaling, so baseview should treat it
-            // as part of our logical size
+            // as part of our logical size. This call with trigger a baseview window resize event,
+            // where the actual stored sizes are updated.
             window.resize(baseview::Size {
                 width: self.current_window_size.width as f64 * self.current_user_scale_factor,
                 height: self.current_window_size.height as f64 * self.current_user_scale_factor,
             });
 
-            // TODO: These calculations are now repeated in three places, should probably be moved
-            //       to a function
+            // TODO: Without this `WindowEvent::GeoChanged` isn't emitted for every element, even
+            //       though this same function is also called in the
+            //       `baseview::WindowEvent::Resized` event handler. Why?
             cx.set_scale_factor(self.window_scale_factor * self.current_user_scale_factor);
```

Comment:
This is the part I don't get. This call shouldn't be necessary since the event handler calls the same function with the same values, but without it not all `GeometryChanged` events are fired. Specifically, the event is not fired for `Model`s, like in the `user_scale` example.

Reply:
Something similar happens when changing just the window size (and this call doesn't help there). `GeometryChanged` isn't called for `Model`s when changing just the window size.